lib/common/s3.sh

```diff
@@ -89,3 +89,53 @@ s3_ls() {
     $S3CMD ls $S3_BUCKET/$path/ | tr -s " " | cut -d' ' -f4 | sed -e "s#^$S3_BUCKET/$path/##"
 }
 export -f s3_ls
+
+# list recursively a folder on s3 and returns list of fullpath of file objects
+# $1 - path of folder on s3
+s3_ls_recur() {
+    local folder=${1%/}; shift # remove trailing slash
+    local object
+
+    for object in `s3lsv -a -b imos-data -k $folder/`; do
+        # check if object is a folder
+        if `echo "$object" | grep -E -q '\/$'`; then
+            s3_ls_recur "$object"
+        else
+            echo "$object" | grep -o -E "$folder.*$"
```

Comment:
I think it would be more up to the s3lsv python function https://github.com/aodn/chef/blob/master/cookbooks/imos_po/files/default/s3lsv to handle this with a different optional argument to only list objects which exists(visible). Doing this here would be patchy and couldn't be re-use in my opinion while it would actually be great to have it inside s3lsv